hicommonwealth/commonwealth
# Update thread list when topics change rapidly

## Link to Issue
Closes: https://github.com/hicommonwealth/commonwealth/issues/4254

## Description of Changes
Now refreshing the `Virtuoso` list component whenever thread topic / stage / date-range / featured-filter changes from either the filter row or from the community sidebar or from url.

## Test Plan
It was a little difficult to work with in local due to less data but these steps should work

1. Go to a community with minimum 1500-2000 threads for a topic (the more the better), select some topics and scroll to load a decent amount of topics > 500
2. Slowly switch between multiple topics from the sidebar (make sure the topics have a lot of threads) - this will make sure we save those threads in local state
3. Now rapidly switch b/w the same topics, this time the topics from the state will be used.
4. Verify the thread list section on the page, it should not get stuck. If it takes time a placeholder bobber will be shown in place of thread cards.

## Deployment Plan
N/A

## Other Considerations
N/A
